cloud_utils/storage/azure.py

```diff
@@ -66,7 +67,9 @@ def head_headers_and_url(bucket_name: str, blob_name: str):
 
 def _blob_service():
     return blob.BlockBlobService(
-        connection_string=os.getenv("AZURE_STORAGE_CONNECTION_STRING"),
```

Comment:
Did you check that the environment variable doesn't contain these characters ? Doesn't sound like the right solution here.